cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -76,17 +74,28 @@ func NewKubeWaiter(client clientset.Interface, timeout time.Duration, writer io.
 
 // WaitForAPI waits for the API Server's /healthz endpoint to report "ok"
 func (w *KubeWaiter) WaitForAPI() error {
+	fmt.Println("[api-check] Waiting for a healthy API server")
+
 	start := time.Now()
-	return wait.PollImmediate(kubeadmconstants.APICallRetryInterval, w.timeout, func() (bool, error) {
-		healthStatus := 0
-		w.client.Discovery().RESTClient().Get().AbsPath("/healthz").Do(context.TODO()).StatusCode(&healthStatus)
-		if healthStatus != http.StatusOK {
-			return false, nil
-		}
+	err := wait.PollUntilContextTimeout(
+		context.Background(),
+		kubeadmconstants.APICallRetryInterval,
+		w.timeout,
+		true, func(_ context.Context) (bool, error) {
```

Comment:
```suggestion
		true, func(ctx context.Context) (bool, error) {
```

Reply:
Maybe we should pass `context.Context` func param from top caller.
That can be considered as a long-term restructuring plan.

Reply:
also applying a context in WaitForKubelet with http.NewRequestWithContext() and client.Do()